QEfficient/transformers/quantizers/quantizer_gptq.py

```diff
@@ -0,0 +1,152 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+
+import torch
+import tqdm
+from transformers.quantizers.quantizer_gptq import HfQuantizer
+from transformers.utils.quantization_config import GPTQConfig
+
+from QEfficient.transformers.quantizers.gptq import QuantLinearGPTQ
+from QEfficient.transformers.quantizers.qunatizer_utils import (
+    find_layers,
+    get_keys_to_not_convert,
+    replace_linear_layer_with_target_layer,
+)
+from QEfficient.utils.logging_utils import logger
+
+
+class QEffGPTQConfig(GPTQConfig):
+    """
+    Configuration class for QEffGPTQ, extending GPTQConfig.
+    This class includes a post-initialization safety checker to ensure that the configuration arguments are correct.
+    """
+
+    def post_init(self):
+        r"""
+        Safety checker that arguments are correct.
+        """
+        if self.bits != 4:
+            raise ValueError(f"Only 4-bit quantization is supported, got bits={self.bits}")
+        if self.desc_act:
+            raise ValueError("Only GPTQ model without decreasing activation size supported.")
```

Comment:
What is decreasing activation? do you mean descending?

Reply:
Yes.